### ISSUE TITLE
Use Round-tripping with Visual Studio

### ISSUE THREAD

**Issue by micove** [PCSX2/pcsx2#684]
Hello:

Visual Studio 2012 introduced what MS calls round-tripping:
http://blogs.msdn.com/b/visualstudio/archive/2012/03/28/round-tripping-with-visual-studio-11.aspx
This feature got backported to VS 2010 in VS 2010 SP1.

Basically a single solution can be used seamlessly for any version above the one in the solution file.
```
MinimumVisualStudioVersion = 10.0.40219.1
```
This one is 2010 SP1 so at least the vs2013.sln works with VS2010 SP1 and up.

The changes to accomplish a unified solution are simple:
1. Check if the vs2012 and vs2013 are currently identical. They should be identical but merge the difference if they are needed.
2. Rename the VS files to something w/o VSXXXX and delete the versioned versions.
3. Change the toolset in the vcxproj file. Basically
```
<Project DefaultTargets="Build" ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
to
```
<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
As explained in https://msdn.microsoft.com/en-us/library/Bb383985.aspx
```
The ToolsVersion attribute of the Project element of the project file. If this attribute doesn’t exist, it is assumed to be the current version.
```
4. Don't hardcode the PlatformToolset in the project files. This one is tricky since currently PCSX2 builds with XP support which:
    - 2010 SP1 = It was included in v100
    - 2012 = It was included in v110_xp (installed by default)
    - 2013 = It was included in v120_xp (installed by default)
    - 2015 = It was included in v140_xp (not installed by default)

XP support is probably going to be gone in the next revision so at some point it will have to be removed.

Anyway to fix this change (using the vs2013.proj as an example)

```
<PlatformToolset>v120_xp</PlatformToolset>
```

to either:

No XP
```
<PlatformToolset>$(DefaultPlatformToolset)</PlatformToolset>
```

With XP and hope for eternal _xp support (most likely removed in VS NEXT)
```
<PlatformToolset Condition="'$(VisualStudioVersion)' != '10.0'">$(DefaultPlatformToolset)_xp</PlatformToolset>
```
This picks up v100 in 2010 and vs1XX_xp for the rest.

Alternative the more verbose version for XP
```
<PlatformToolset Condition="'$(VisualStudioVersion)' == '11.0' or '$(VisualStudioVersion)' == '12.0' or '$(VisualStudioVersion)' == '14.0'">$(DefaultPlatformToolset)_xp</PlatformToolset>
<PlatformToolset Condition="'$(VisualStudioVersion)' != '11.0' and '$(VisualStudioVersion)' != '12.0' and '$(VisualStudioVersion)' != '14.0'">$(DefaultPlatformToolset)</PlatformToolset>
```
This one will use vs110_xp, vs120_xp, vs140_xp and for the rest it will use the default one v100/v150/v160.

I have a unpushed branch with this but I'm asking before I do a PR. A decision about XP would also be helpful. I would drop XP since it hit EOL mainstream/extended already, no security support, etc. This also as a side-effect drops the DirectX SDK dependency for all but GsDX and SPU2-X which need porting.

I tested all of this with VS2013/VS2015 and everything compiled using the same solution w/o any "please upgrade the projects" window. It should work on VS2012 too but c++11 support in that version is laughable and I don't really want to install it.

**Comment by gregory38** [comment]
Fwiw VS2010 support was dropped (due to C++11). However I'm in favor to keep XP support. Sure it is EOL but lots of people still use it. http://gs.statcounter.com/#desktop-os-ww-monthly-201406-201507 . We will see next year. Hopefully users will have enough money to upgrade to win10 (or switch to linux). Anyway it is a decision for the windows developer.

**Comment by micove** [comment]
Well I have a branch here:
https://github.com/micove/pcsx2/tree/VS_universal_sln

I can redo it with XP support since it's trivial.

I could also remove the 2012 files and I did not upgrade the files of the vs2013 projects that are not in the solution.

I also noticed that currently opencl compiles using v120 instead of v120_xp. Also opencl is not in the 2012 solution but the 2012 files are usually less updated than the vs2013 ones so I based everything on the 2013 files.
